File: 1762-buildings-with-an-ocean-view/1762-buildings-with-an-ocean-view.cpp
class Solution {
public:
    vector<int> findBuildings(vector<int>& heights) {
        int maxV = -1;
        vector<int> ans;
        
        for(int i = heights.size() - 1; i >= 0; i--)
        {
            if(heights[i] > maxV)
            {
                ans.push_back(i);
            }
            
            maxV = max(maxV, heights[i]);
        }
        
        reverse(ans.begin(), ans.end());
        
        return ans;
    }
};

File: 1762-buildings-with-an-ocean-view/1762-buildings-with-an-ocean-view.py
class Solution:
    def findBuildings(self, heights: List[int]) -> List[int]:
        res = []
        maxH = 0
        
        for i in range(len(heights) - 1, -1, -1):
            if heights[i] > maxH:
                res.append(i)
                
                maxH = heights[i]
        
        return res[::-1]